source/particle/world.cc

```diff
@@ -617,6 +614,158 @@ namespace aspect
 
     namespace internal
     {
+
+      /**
+       * Wrapper around dealii::FEPointEvaluation to choose number of components dynamically.
+       */
+      template <int dim>
+      class DynamicFEPointEvaluation
+      {
+        public:
+          DynamicFEPointEvaluation(const unsigned int first_component, const unsigned int n_components)
+            : first_component (first_component),
+              n_components(n_components)
+          {}
+
+          virtual ~DynamicFEPointEvaluation() = default;
+
+          int first_component;
+          int n_components;
+
+          virtual void evaluate(const ArrayView<double> &solution_values,
+                                const EvaluationFlags::EvaluationFlags flags) = 0;
+
+          virtual
+          std::vector<Tensor<1,dim>>
+          get_gradient(const unsigned int evaluation_point) const = 0;
+
+          virtual
+          std::vector<double>
+          get_value(const unsigned int evaluation_point) const = 0;
+
+      };
+
+
+
+      template <int dim>
+      Tensor<1,dim> to_tensor(const Tensor<1,dim> &in)
+      {
+        return in;
+      }
```

Comment:
I dont understand the necessity for `to_tensorx` functions. At first I thought you use them to create a non-const copy, but in the place they are used below they are not modified. Can you explain and document?

Edit: Oh, do you need the specialization for `dim` == 1 because the type is different? That would be something to document.

Reply:
Exactly. This is documented now.